### PR TITLE
Override sysinfo syscall handling

### PR DESCRIPTION
**Pull request by jxyang**

### CODE REVIEW

**Comment by davidchisnall** [comment]
Looks fine, please can you also remove the musl syscall bypass and bump the sub module?

**Comment by davidchisnall** [comment]
Sorry, I didn’t see that you’d already sent the musl PR. Please bump the sub module so that CO can run with the new libc.

File: src/enclave/enclave_mem.c
Comment (davidchisnall) [inline src/enclave/enclave_mem.c]: This looks like a type name without a variable name.  What does this actually do?

File: src/include/enclave/enclave_mem.h
Comment (davidchisnall) [inline src/include/enclave/enclave_mem.h]: Please can you pull the overridden syscalls' prototypes into a separate header and move the implementation into the `src/lkl/` directory?  These are LKL-specific, not generic support code.

File: src/lkl/syscall-overrides.c
Comment (davidchisnall) [inline src/lkl/syscall-overrides.c]: I think this is the wrong syscall number.  LKL has different syscall numbers to x86-64 Linux.  This needs to be `__lkl__NR_sysinfo`.  See the changes for the futex override in #162 for how to structure this.

**Review by davidchisnall — CHANGES_REQUESTED** [review]

File: src/enclave/enclave_mem.c
Comment (jxyang) [reply src/enclave/enclave_mem.c]: This just declares to the compiler `struct sysinfo` is a type, so it doesn't complain anything like `struct sysinfo*`.

File: src/lkl/syscall-overrides.c
Comment (jxyang) [reply src/lkl/syscall-overrides.c]: `SYS_sysinfo` has the same value as `__NR_sysinfo`. I couldn't find anywhere we define `__lkl__NR_sysinfo` though.

**Comment by jxyang** [comment]
@davidchisnall Is there a way to bump the submodule before the sgx-lkl-musl change is merged?

File: src/lkl/syscall-overrides.c
Comment (jxyang) [reply src/lkl/syscall-overrides.c]: In any case, I will update to `__lkl__NR_sysinfo` as the debugger shows it also has the same value.

File: src/include/enclave/enclave_mem.h
Comment (jxyang) [reply src/include/enclave/enclave_mem.h]: The overriding function uses static variables from `enclave_mem.c`. What's the recommended way to share variables between `src/lkl` and `src/enclave`?

File: src/enclave/enclave_mem.c
Comment (davidchisnall) [reply src/enclave/enclave_mem.c]: Please remove the `static` here (it doesn't mean anything and I suspect it's not actually valid C).  The entire line is unnecessary though, because it's a forward declaration that comes after the real declaration.  Fields of this struct are used after this line with no other `sysinfo` declaration after this point.  This line should be removed.

Actually, after your most recent change, `sysinfo` isn't used in this file, so the forward declaration isn't needed.

File: src/include/lkl/syscall-overrides-sysinfo.h
Comment (davidchisnall) [inline src/include/lkl/syscall-overrides-sysinfo.h:9]: Please add a doc comment.

File: src/include/enclave/enclave_mem.h
Comment (davidchisnall) [inline src/include/enclave/enclave_mem.h:37]: Please add a doc comment.

File: src/lkl/syscall-overrides-sysinfo.c
Comment (davidchisnall) [inline src/lkl/syscall-overrides-sysinfo.c]: Please add a comment that this should report the number of ethreads.

**Review by davidchisnall — CHANGES_REQUESTED** [review]

File: src/include/lkl/syscall-overrides-sysinfo.h
Comment (jxyang) [reply src/include/lkl/syscall-overrides-sysinfo.h:9]: Done

**Comment by jxyang** [comment]
@davidchisnall All comments are addressed. Please review again.

**Review by davidchisnall — APPROVED** [review]

**Comment by davidchisnall** [comment]
@jxyang, please rebase this so that it can be merged?

**Comment by davidchisnall** [comment]
I have done the rebase and cleanup, this is now ready to merge once CI is happy.

**Pull request merged**
